### ISSUE TITLE
Change HashTable Parameter Map to Allow Null Values

### ISSUE THREAD

**Issue by volosied** [OpenLiberty/open-liberty#16589]
Currently, the webcontainer parameter map  is a HashTable.  Per the JavaDoc:
`This class implements a hash table, which maps keys to values. Any non-null object can be used as a key or as a value. `

This means that a request like http://localhost:9080/servlet/HelloWorld?age=15&name does not keep track of the name key in the map since no value is assigned.  (One workaround is to use "age=", however. Other is to set  _allowQueryParamWithNoEqual_ to true)

Per the JavaDoc: Map<String,String[]> getParameterMap() 
 - https://javaee.github.io/javaee-spec/javadocs/javax/servlet/ServletRequest.html#getParameterMap--
 - A Map must be returned, but, as mentioned, our underlying data structures HashTable. 
 
 If a developer wanted check the keys for the request above, _age_ would be NOT be listed.   
 
 Basically, we would need to change the HashTable to something else.   Perhaps a HashMap?  Or a custom class that implements the Map interface? 
 
 We could simply use _allowQueryParamWithNoEqual_ but that might affect application behavior. 
 
 This issue has been spurred by a recent TCK failure in JAX-RS.
